pyproject.toml

```diff
@@ -35,6 +35,7 @@ dependencies = [
   "attrs",
   "pandas",
   "cloudpathlib",
+  "tenacity"
```

Comment:
Let's keep # of dependencies low? I think we can pretty much rewrite this.

Reply:
Based on internal discussions decided to keep